### PR TITLE
chore: cloudflare cache expose cachebuster in constructor

### PR DESCRIPTION
**Pull request by MichaelUnkey**
## What does this PR do?

Eng-1079

Fixes # (issue)

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] Chore (refactoring code, technical debt, workflow improvements)
* [ ] Enhancement (small improvements)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## How should this be tested?

* Test A
* Test B

### Required

* [ ] Filled out the "How to test" section in this PR
* [ ] Read [Contributing Guide](./CONTRIBUTING.md)
* [ ] Self-reviewed my own code
* [ ] Commented on my code in hard-to-understand areas
* [ ] Ran `pnpm build`
* [ ] Ran `pnpm fmt`
* [ ] Checked for warnings, there are none
* [ ] Removed all `console.logs`
* [ ] Merged the latest changes from main onto my branch with `git pull origin main`
* [ ] My changes don't cause any responsiveness issues

### Appreciated

* [ ] If a UI change was made: Added a screen recording or screenshots to this PR
* [ ] Updated the Unkey Docs if changes were necessary

<sub><a href="https://huly.app/guest/unkey?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2Njg1NTRmOTJhNmFiOWVlZjU1YTAxOTEiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctamFtZXMtdW5rZXktNjY2YWYwNzktNjliNzJiMzYyNi05ZGM2MGUiLCJwcm9kdWN0SWQiOiIifQ.ZscVilzoD5pwSBfnHjfJMzl2ZQ3VW7AUPB3Fl9IGFoc">Huly&reg;: <b>ENG-1847</b></a></sub>

### CODE REVIEW

**Review by chronark — CHANGES_REQUESTED** [review]
Please make the cacheBuster an optional string, fall back to `v1` and remove it from all the examples, except documentation

**Comment by MichaelUnkey** [comment]
What part did I miss?

**Comment by chronark** [comment]
`cacheBuster: string | "v1"`  is not optional
what you need to do is `cacheBuster?: string`
and then in the code you need to fall back to `"v1"`  if it’s undefined

**Comment by MichaelUnkey** [comment]
got it

**Comment by MichaelUnkey** [comment]
Should I add the cachebuster to all examples on this page of docs or just in cloudflare section? For example the quickstart at the top? [https://www.unkey.com/docs/libraries/ts/cache/overview#quickstart](https://www.unkey.com/docs/libraries/ts/cache/overview#quickstart)
I would guess no cause its optional but not 100% on this

**Comment by chronark** [comment]
just in the section I linked
[https://www.unkey.com/docs/libraries/ts/cache/overview#cloudflare](https://www.unkey.com/docs/libraries/ts/cache/overview#cloudflare)

File: packages/cache/src/stores/cloudflare.ts [hunk 1]
```diff
@@ -10,6 +10,11 @@ export type CloudflareStoreConfig = {
    * This token must have at least
    */
   cloudflareApiKey: string;
+
+  /**
+   * Set for cachestore version
```
Comment (chronark) [inline packages/cache/src/stores/cloudflare.ts:15]:
```suggestion
   * The version of the cache store schema. If changes occur on your cache schema, 
   * incrementing this value ensures the old cached schema is not utilized.
   * @default "v1"
```

**Comment by chronark** [comment]
Here’s what I think needs to be included in the docs:
1. when do I need this
2. what does it do
3. Where is the value used

How about something along the lines of:
>As your data changes, it is important to keep backwards compatibility in mind. If your cached values are no longer backwards compatible, it can cause problems. For example when a value changes from optional to required. In these cases you should purge the entire cache by setting a new `cacheBuster`  value. The `cacheBuster`  is used as part of the cache key and changes ensure you are not reading old data anymore.

Defaults to `v1`  if omitted.

**Comment by MichaelUnkey** [comment]
Sounds good to me. I need to learn to elaborate more .

**Pull request merged**